frappe/utils/global_search.py

```diff
@@ -373,11 +373,57 @@ def sync_global_search():
 	:param flags:
 	:return:
 	"""
-	while frappe.cache.llen("global_search_queue") > 0:
-		# rpop to follow FIFO
-		# Last one should override all previous contents of same document
-		value = json.loads(frappe.cache.rpop("global_search_queue").decode("utf-8"))
-		sync_value(value)
+	from itertools import islice
+
+	def get_search_queue_item_generator():
+		while frappe.cache.llen("global_search_queue") > 0:
+			yield frappe.cache.rpop("global_search_queue")
+
+	item_generator = get_search_queue_item_generator()
+	while search_items := tuple(islice(item_generator, 10_000)):
+		values = get_deduped_search_item_values(search_items)
+		sync_values(values)
+
+
+def get_deduped_search_item_values(items):
+	from collections import OrderedDict
+
+	values_dict = OrderedDict()
+	for item in items:
+		item_json = item.decode("utf-8")
+		item_dict = json.loads(item_json)
+		key = (item_dict["doctype"], item_dict["name"])
+
+		if key in values_dict:
+			del values_dict[key]
+
```

Comment:
```suggestion
```

Not required? You're overriding the key in next line.

Reply:
<img width="308" alt="Screenshot 2023-11-06 at 11 49 14" src="https://github.com/frappe/frappe/assets/29507195/7a107acd-db24-41ac-bf80-7f9fd70ee7d9">

If insertion order is unimportant then yeah, not required.

Considering the [specific use of `rpop`](https://github.com/frappe/frappe/pull/23064/files/3d7ba41583ffabdddef007a9a7274024ab305c96#diff-2df9e640d5d5f2f19851093522f542bbdceea96852667ee33042ad22caa95ac5L377) I thought it would be.

Reply:
@18alantom we just need the last write to win, it doesn't have to actually be in order. So we are fine I think. 

Also, python ordereddict isn't really required too, dict by default is ordered too (enough for this use case) :smile: